test/unit/SourceLocator/Type/AnonymousClassObjectSourceLocatorTest.php

```diff
@@ -0,0 +1,122 @@
+
+            }
+
+        };
+    }
+
+    public function testLocateIdentifier() : void
+    {
+        $class = $this->getAnonymousClassOne();
+
+        /** @var ReflectionClass $reflection */
+        $reflection = (new AnonymousClassObjectSourceLocator($class))->locateIdentifier(
+            $this->createMock(Reflector::class),
+            new Identifier(
+                get_class($class),
+                new IdentifierType(IdentifierType::IDENTIFIER_CLASS)
+            )
+        );
+
+        self::assertTrue($reflection->isAnonymous());
+        self::assertStringStartsWith(ReflectionClass::ANONYMOUS_CLASS_NAME, $reflection->getShortName());
+        self::assertSame(21, $reflection->getStartLine());
+        self::assertSame(28, $reflection->getEndLine());
+    }
+
+    public function testLocateIdentifierWithFunctionIdentifier() : void
+    {
+        /** @var ReflectionClass|null $reflection */
+        $reflection = (new AnonymousClassObjectSourceLocator($this->getAnonymousClassOne()))->locateIdentifier(
+            $this->createMock(Reflector::class),
+            new Identifier(
+                'foo',
+                new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION)
+            )
+        );
+
+        self::assertNull($reflection);
+    }
+
+    public function testLocateIdentifiersByType() : void
+    {
+        $class = $this->getAnonymousClassTwo();
+
+        /** @var ReflectionClass[] $reflections */
+        $reflections = (new AnonymousClassObjectSourceLocator($class))->locateIdentifiersByType(
+            $this->createMock(Reflector::class),
+            new IdentifierType(IdentifierType::IDENTIFIER_CLASS)
+        );
+
+        self::assertCount(1, $reflections);
+        self::assertArrayHasKey(0, $reflections);
+
+        self::assertTrue($reflections[0]->isAnonymous());
+        self::assertStringStartsWith(ReflectionClass::ANONYMOUS_CLASS_NAME, $reflections[0]->getShortName());
+        self::assertSame(33, $reflections[0]->getStartLine());
+        self::assertSame(40, $reflections[0]->getEndLine());
+    }
+
+    public function testLocateIdentifiersByTypeWithFunctionIdentifier() : void
+    {
+        $class = $this->getAnonymousClassTwo();
+
+        /** @var ReflectionClass[] $reflections */
+        $reflections = (new AnonymousClassObjectSourceLocator($class))->locateIdentifiersByType(
+            $this->createMock(Reflector::class),
+            new IdentifierType(IdentifierType::IDENTIFIER_FUNCTION)
+        );
+
+        self::assertCount(0, $reflections);
+    }
+
+    public function testExceptionIfTwoAnonymousClassesOnSameLine() : void
+    {
+        self::expectException(TwoAnonymousClassesOnSameLine::class);
+        self::expectExceptionMessage(sprintf('Two anonymous classes on line 113 in %s', FileHelper::normalizeWindowsPath(__FILE__)));
+
+        $class1 = new class {}; $class2 = new class {};
+       (new AnonymousClassObjectSourceLocator($class1))->locateIdentifier(
```

Comment:
Wouldn't this test parse the test class itself? I think it's more efficient to split out this line:
```php
<?php
$class1 = new class {}; $class2 = new class {};
return [$class1, $class2];
```
Into a separate file, for ease of testing/parsing